test/data/core/types/CoreAddress.ts

```diff
@@ -15,21 +15,27 @@ export class CoreAddress extends CoreSerializable {
     @serialize()
     public address: string
 
-    public static from(value: ICoreAddress | string): CoreAddress {
+    protected static preFrom(value: any): any {
         if (typeof value === "string") {
-            return super.from({ address: value }, CoreAddress) as CoreAddress
+            return { address: value }
         }
-        return super.from(value, CoreAddress) as CoreAddress
+
+        return value
     }
 
-    public static deserialize(value: string): CoreAddress {
-        try {
-            return super.deserializeT<CoreAddress>(value, CoreAddress)
-        } catch (e) {
-            return this.from(value)
-        }
+    public static from(value: ICoreAddress | string): CoreAddress {
+        return this.fromAny(value)
     }
 
+    // TODO: ??
```

Comment:
Zugegeben, total verwirrend. Die Idee dahinter war, dass das Deserialisieren eines Strings bei einem Datentyp, der nur ein String ist, doppeltgemoppelt ist. Das heißt, ein Developer könnte entweder deserialize aufrufen, weil es sieht, das ist ein String und a) es hat tatsächlich ein serialisiertes Objekt '{"@type":"CoreAddress","address":"blubb"}') oder b) es hat nur eine Adresse "blubb". Man weiß es ja in dem Fall dummerweise nicht. Daher erst das Deserialisieren probieren, danach ein einfaches from.